### PR TITLE
Add go-md2man to repository

### PR DESCRIPTION
**Pull request by saschagrunert**
Hey, to avoid changing dependencies which could break our builds (like the travis build in #2182) I would go the way to vendor necessary test dependencies directly into the repository.

I did this for go-md2man within this PR and cleaned up the Makefile a bit. If you're fine with that I would do the same for the rest of the external dependencies.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -155,21 +155,35 @@ dbuild: crioimage
...
+	$(call go-build,mockgen,./vendor/github.com/golang/mock/mockgen)
 
 vendor: .install.vndr
 	$(GOPATH)/bin/vndr \
-		-whitelist "github.com/onsi/ginkgo" \
-		-whitelist "github.com/golang/mock" \
+		-whitelist github.com/onsi/ginkgo \
+		-whitelist github.com/golang/mock \
+		-whitelist github.com/cpuguy83/go-md2man \
+		-whitelist github.com/russross/blackfriday \
```
Comment (TomSweeneyRedHat) [inline Makefile]: are you using blackfriday with this PR?  Or is this something you need in place for later on?

Comment (saschagrunert) [reply Makefile]: Yes we need it as direct dependency of `go-md2man` and have to whitelist it there to avoid removal during running `make vendor`.

Comment (TomSweeneyRedHat) [reply Makefile]: OK, maybe we're calling/using go-md2man in Podman and Buildah differently, but we're not pulling in blackfriday in either of those projects that I can see.

**Comment by TomSweeneyRedHat** [comment]
LGTM, assuming happy tests

**Comment by haircommander** [comment]
@saschagrunert rebasing should fix CI issues :)

**Comment by haircommander** [comment]
Tide is not cooperating. Merging manually

**Pull request merged**